### PR TITLE
Fix typing hints in vectorized_mobject.py based on mypy 

### PR DESCRIPTION
**Pull request by cobordism**
## Changelog / Overview
Based on the program outlined in #1212 I have started fixing errors showing up in mypy.

The only real changes in this PR are type hints and so do not affect the runtime of manim at all.

## Motivation
I want to get `mypy -m manim` down to zero errors.

## Explanation for Changes
The mypy static type checker finds hundreds of 'errors' in our code. Most of the time these are incorrect type hints, or just sloppy formatting.  This PR fixes a number of mypy errors coming out of vectorized_mobject.py

## Testing Status
Functional code is unchanged by this PR.

## Checklist
- [ ] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [x] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Comment by naveen521kk** [comment]
> I want to get mypy -m manim down to zero errors.

Once it does we can have a check for that! Thanks for your contribution.

File: manim/utils/bezier.py [hunk 1]
```diff
@@ -108,7 +108,9 @@ def interpolate(start: int, end: int, alpha: float) -> float:
     return (1 - alpha) * start + alpha * end
 
 
-def integer_interpolate(start: float, end: float, alpha: float) -> int:
+def integer_interpolate(
+    start: float, end: float, alpha: float
+) -> typing.Tuple(int, float):
```
Comment (naveen521kk) [inline manim/utils/bezier.py:113]:
```suggestion
) -> typing.Tuple[int, float]:
```

Comment (naveen521kk) [reply manim/utils/bezier.py:113]: oh wait that should be wrong.

**Review by jsonvillanueva — APPROVED** [review]
LGTM! Just settle the merge conflict(s) and feel free to merge.

**Review by naveen521kk — APPROVED** [review]
lgtm

File: manim/mobject/types/vectorized_mobject.py [hunk 2]
```diff
@@ -621,7 +621,7 @@ def add_points_as_corners(self, points: np.ndarray) -> "VMobject":
             self.add_line_to(point)
         return points
 
-    def set_points_as_corners(self, points: Iterable[float]) -> "VMobject":
+    def set_points_as_corners(self, points: Sequence[float]) -> "VMobject":
```
Comment (kolibril13) [inline manim/mobject/types/vectorized_mobject.py:624]: I think that `Sequence[float]` is not the right type to choose here.
1. Would a sequence of int not be fine as well?
2. Why is the type hint not e.g. Union[np.ndarray, List], as also here:
https://github.com/ManimCommunity/manim/blob/808c35fa3f31a1dc98ee87f11567ff9359ccb1b5/manim/mobject/mobject.py#L2006

Comment (cobordism) [reply manim/mobject/types/vectorized_mobject.py:624]: hey.
All I could tell was that Iterable was wrong and Sequence was "better".

There is still a lot of work to be done to fix and polish all of our type hints. I will note down for the next time to take another look at this.

**Comment by naveen521kk** [comment]
@cobordism Please squash while merging or rebase this PR before merging so that it doesn't contain merge commits https://github.com/ManimCommunity/manim/compare/808c35fa3f31...227dd5548720.

**Comment by cobordism** [comment]
Apologies, I was under the impression we had long ago set squash+merge to happen automatically..

**Comment by naveen521kk** [comment]
> Shall I revert?

I think it isn't possible to revert without a force push to master, so we can't do anything let it be as is.

**Pull request merged**